test/state/precompiles.cpp

```diff
@@ -292,6 +293,35 @@ ExecutionResult blake2bf_execute(const uint8_t* input, [[maybe_unused]] size_t i
     return {EVMC_SUCCESS, sizeof(h)};
 }
 
+namespace
+{
+using intx::operator""_u256;
+constexpr auto FIELD_ELEMENTS_PER_BLOB = 4096_u256;
+constexpr auto BLS_MODULUS =
+    52435875175126190479447740508185965837690552500527637822603658699938581184513_u256;
+}  // namespace
+
+ExecutionResult point_evaluation_execute(const uint8_t* input, size_t input_size, uint8_t* output,
+    [[maybe_unused]] size_t output_size) noexcept
+{
+    assert(output_size >= 64);
+    if (input_size != 192)
+        return {EVMC_PRECOMPILE_FAILURE, 0};
+
+    const auto r = crypto::kzg_verify_proof(reinterpret_cast<const std::byte*>(&input[0]),
+        reinterpret_cast<const std::byte*>(&input[32]),
+        reinterpret_cast<const std::byte*>(&input[64]),
+        reinterpret_cast<const std::byte*>(&input[96]),
+        reinterpret_cast<const std::byte*>(&input[96 + 48]));
+
+    if (!r)
+        return {EVMC_PRECOMPILE_FAILURE, 0};
+
+    intx::be::unsafe::store(output, FIELD_ELEMENTS_PER_BLOB);
+    intx::be::unsafe::store(output + 32, BLS_MODULUS);
+    return {EVMC_SUCCESS, 64};
+}
```

Comment:
For this, could we link to the eip method or explain what the offsets are being used for?

Reply:
Done.